e3sm_diags/driver/utils/climo_xr.py

```diff
@@ -137,15 +148,48 @@ def climo(dataset: xr.Dataset, var_key: str, freq: ClimoFreq):
     time_bnds = ds.bounds.get_bounds(axis="T")
     time_lengths = (time_bnds[:, 1] - time_bnds[:, 0]).astype(np.float64)
 
-    # Calculate the weighted average of the masked data variable using the
-    # appropriate indexes and weights.
-    climo = ma.average(dv_masked[time_idx], axis=0, weights=time_lengths[time_idx])
+    ncycle = len(cycle)
+    climo = ma.zeros([ncycle] + list(np.shape(dv))[1:])
+    for n in range(ncycle):
+        # print("cycle:", cycle[n], len(time_coords))
+        # Loop over the time coordinates to get the indexes related to the
+        # user-specified climatology frequency using the frequency index map
+        # (`FREQ_IDX_MAP``).
+        # Using a list comprehension to make looping faster, also
+        # to have time_coords_months an array gets more speedup.
+
+        time_idx = np.array(
+            [
+                FREQ_IDX_MAP[cycle[n]][time_coords_months[i] - 1]
+                for i in range(len(time_coords_months))
+            ],
+            dtype=np.int64,
+        ).nonzero()
+
+        # time_idx = []
+        # for i in range(len(time_coords)):
+        #    month = time_coords[i].dt.month.item()
+        #    idx = FREQ_IDX_MAP[cycle[n]][month - 1]
+        #    time_idx.append(idx)
+
+        # time_idx = np.array(time_idx, dtype=np.int64).nonzero()  # type: ignore
+
+        # Calculate the weighted average of the masked data variable using the
+        # appropriate indexes and weights.
+        climo[n] = ma.average(
+            dv_masked[time_idx], axis=0, weights=time_lengths[time_idx]
+        )
```

Comment:
I don't think these code optimizations in `climo_xr.py` are necessary anymore with the `.load()` solution mentioned [here] https://github.com/E3SM-Project/e3sm_diags/pull/842#issuecomment-2339217692).